Modules/posixmodule.c

```diff
@@ -3869,7 +3880,18 @@ posix_fork(PyObject *self, PyObject *noargs)
     pid_t pid;
     int result = 0;
     _PyImport_AcquireLock();
+#ifdef WITH_THREAD
+    if (_PyThread_AcquireKeyLock() == 0) {
```

Comment:
You should add a comment explaining the rationale of this lock. I understand that it's a workaround to make the Python TLS implementation "fork-safe".